src/main/kotlin/com/cosmotech/api/utils/KubernetesClient.kt

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.api.utils
+
+import io.kubernetes.client.openapi.ApiException
+import io.kubernetes.client.openapi.apis.CoreV1Api
+import io.kubernetes.client.openapi.models.V1ObjectMeta
+import io.kubernetes.client.openapi.models.V1Secret
+import io.kubernetes.client.util.ClientBuilder
+import java.util.Base64
+import org.slf4j.LoggerFactory
+import org.springframework.context.annotation.Bean
+import org.springframework.stereotype.Service
+
+@org.springframework.context.annotation.Configuration
+open class KubernetesApi {
+  @Bean
+  open fun coreV1Api(): CoreV1Api {
+    val client = ClientBuilder.defaultClient()
+    return CoreV1Api(client)
+  }
+}
+
+@Service
+class KubernetesClient(private val kubernetesApi: CoreV1Api) : SecretManager {
+
+  private val logger = LoggerFactory.getLogger(KubernetesClient::class.java)
+
+  override fun createSecret(
+      tenantName: String,
+      secretName: String,
+      secretData: Map<String, String>
+  ) = this.createSecretIntoKubernetes(tenantName, secretName, secretData)
+
+  override fun readSecret(tenantName: String, secretName: String): Map<String, String> =
+      this.getSecretFromKubernetes(tenantName, secretName)
+
+  private fun getSecretFromKubernetes(
+      namespace: String = "phoenix",
+      secretName: String
+  ): Map<String, String> {
+    val result = kubernetesApi.readNamespacedSecret(secretName, namespace, "")
+
+    logger.debug("Secret retrieved")
+    return result.data?.mapValues { Base64.getDecoder().decode(it.value).toString(Charsets.UTF_8) }
+        ?: mapOf()
+  }
+
+  private fun createSecretIntoKubernetes(
+      namespace: String,
+      secretName: String,
+      secretData: Map<String, String>
+  ) {
+    logger.debug("Creating secret $secretName in namespace $namespace")
+
+    val secretNameLower = secretName.lowercase()
+    val metadata = V1ObjectMeta()
+    metadata.name = secretNameLower
+    metadata.namespace = namespace
+    val body = V1Secret()
+    body.metadata = metadata
+
+    body.data = secretData.mapValues { Base64.getEncoder().encode(it.value.toByteArray()) }
+    body.type = "Opaque"
+
+    var replace = false
+    @Suppress("SwallowedException")
+    try {
+      kubernetesApi.readNamespacedSecret(secretNameLower, namespace, null)
+      logger.debug("Secret $secretNameLower already exists in namespace $namespace: replacing it")
+      replace = true
+    } catch (e: ApiException) {
+      logger.debug("Secret $secretNameLower does not exists in namespace $namespace: creating it")
+    }
+    try {
+      if (replace) {
```

Comment:
I think that we should make a specific public api 'replacesecret' to avoid mixing behavior

Reply:
I craeted a createOrReplace for it to be explicit and simple for usage